File: PythonProjects/guessing_game.py
import random

# Sets a seed for reproducibility
random.seed(10)

# Defines the lower and upper bounds for the random number
lower_bound = 1
upper_bound = 100

# Generates a random number between lower_bound and upper_bound
random_value = random.randint(lower_bound, upper_bound)


def guess_game(seed: int) -> None:
    global random_value

    # Welcome message
    print(
        "Welcome to the Guessing Game!\nI'm thinking of a number between 1 and 100. Can you guess it?"
    )

    # Initializes the number of tries
    no_of_tries = 0

    # Main loop
    while True:
        # Gets user input for the guess
        guess = int(input("Guess a number: "))
        no_of_tries += 1  # Increment the number of tries for each guess

        # Checks if the guess is too low, too high, or correct
        if guess < random_value:
            print("Good try, but that's too low. Try again.")
        elif guess > random_value:
            print("Good try, but that's too high. Try again.")
        else:
            # Displays a congratulatory message when the guess is correct
            print(
                f"Yes! You guessed correctly after {no_of_tries} {'try' if no_of_tries == 1 else 'tries'}! Congratulations."
            )
            break  # Exit the loop when the correct guess is made


# Starts the guessing game with a seed value
guess_game(10)

File: Backup/GuessingGame.java
import java.util.Scanner;
import java.util.Random;

public class GuessingGame {
    public static void main(String[] args) {
        // Starts the guessing game with a seed value of 5
        guessGame(10);
    }

    // Function to play the guessing game
    public static void guessGame(long seed) {
        // Welcome message
        System.out
                .println("Welcome to the Guessing Game!\nI'm thinking of a number between 1 and 100. Can you guess it");

        // Initializes variables
        Scanner input = new Scanner(System.in);
        int guess, no_of_tries = 1, randomValue;

        // Creates a Random object with a specific seed value
        Random random = new Random(10);
        randomValue = random.nextInt(100) + 1; // Generate a random number between 1 and 100
        // System.out.println(randomValue); // Uncomment to display the actual random
        // value

        // Main game loop
        while (true) {
            System.out.println("Guess a number: ");
            guess = input.nextInt();

            // Checks if the guess is too low, too high, or correct
            if (guess != randomValue) {
                no_of_tries++;

                if (guess < randomValue) {
                    System.out.println("Good try, but that's too low. Try again.");
                }

                if (guess > randomValue) {
                    System.out.println("Good try, but that's too high. Try again.");
                }
            } else {
                // Displays a congratulatory message when the guess is correct
                if (no_of_tries == 1)
                    System.out.println("Yes! You guessed correctly after 1 try! Congratulations.");
                else
                    System.out.println("Yes! You guessed correctly after " + no_of_tries + " tries! Congratulations.");

                // Exits the loop when the correct guess is made
                break;
            }
        }

        // Closes the Scanner
        input.close();
    }
}
